### PR TITLE
TST: add instance_of wrapper

### PR DESCRIPTION
**Pull request by llllllllll**

### CODE REVIEW

File: zipline/testing/predicates.py [hunk 1]
```diff
@@ -84,7 +84,37 @@ def __ne__(other):
...
+    def __eq__(self, other):
+        if self.exact:
+            return type(other) in self.types
+
+        return isinstance(other, self.types)
+
+    def __repr__(self):
+        return '<%s: %r%s>' % (
+            type(self).__name__,
+            self.types[0] if len(self.types) == 1 else self.types,
```
Comment (ssanderson) [inline zipline/testing/predicates.py]: This is going to crash on the `len` call if `types` is a single type.

File: zipline/testing/predicates.py [hunk 2]
```diff
@@ -84,7 +84,37 @@ def __ne__(other):
...
+        self.exact = exact
+
+    def __eq__(self, other):
+        if self.exact:
+            return type(other) in self.types
+
+        return isinstance(other, self.types)
+
+    def __repr__(self):
+        return '<%s: %r%s>' % (
```
Comment (ssanderson) [inline zipline/testing/predicates.py]: Is there a reason not to just make this match exactly what the call would look like? `instance_of(str)` seems pretty readable to me?

File: zipline/testing/predicates.py (hunk 1 above)
Comment (ssanderson) [reply zipline/testing/predicates.py]: Oh, nvm, you're converting at construction time.

Comment (llllllllll) [reply zipline/testing/predicates.py]: it is always a tuple, check __init__

File: zipline/testing/predicates.py [hunk 3]
```diff
@@ -84,7 +84,37 @@ def __ne__(other):
...
+
+    Parameters
+    ----------
+    types : type or tuple[type]
+        The types to compare equal to.
+    exact : bool, optional
+        Only compare equal to exact instances, not instances of subclasses?
+    """
+    def __init__(self, types, exact=False):
+        if not isinstance(types, tuple):
```
Comment (ssanderson) [inline zipline/testing/predicates.py]: Might be worth asserting `isinstance(types, type)` here?

**Pull request merged**